### PR TITLE
✨ Implement Simple Confirm Widget

### PR DESCRIPTION
**Pull request by SteffenKn**
## What did you change?

⏰ Waiting for https://github.com/process-engine/management_api_contracts/pull/15
⏰ Waiting for https://github.com/process-engine/management_api_client/pull/10
⏰ Waiting for https://github.com/process-engine/process_engine/pull/142
⏰ Waiting for https://github.com/process-engine/skeleton/pull/72

This PR implements the confirm widget.

### TODO:

- Update Version of skeleton-electron

## How can others test the changes?

- Open up the BPMN-Studio 
- Go to the Detail View of any diagram 
- Add a user task
	- Add a extension property with name "preferredControl" and value "confirm"
	- Add a form field of type boolean with a label
- Run the diagram
- Notice the confirm widget

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [ ] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/confirm-widget/confirm-widget.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  constructor(notificationService: NotificationService) {
+    this._notificationService = notificationService;
+  }
+
+  public attached(): void {
+    const booleanFormField: UserTaskFormField = this._getBooleanFormField();
+    const userTaskHasNoBooleanFormField: boolean = booleanFormField === undefined;
+
+    if (userTaskHasNoBooleanFormField) {
+      const errorMessage: string = 'Confirm User Tasks must have a form field of type boolean that should get confirmed.';
```
Comment (alexanderkasten) [inline src/modules/confirm-widget/confirm-widget.ts]: UserTasks

File: src/modules/confirm-widget/confirm-widget.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import {UserTaskConfig, UserTaskFormField, UserTaskFormFieldType} from '@process-engine/management_api_contracts';
```
Comment (Vyperus) [inline src/modules/confirm-widget/confirm-widget.ts]: Sort imports after this pattern:

1) Frameworkstuff
2) Thirdpartystuff
3) Internstuff

Like we do in all other files.

File: src/modules/form-widget/form-widget.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import {UserTaskFormField, UserTaskFormFieldType} from '@process-engine/management_api_contracts';
+
+import {UserTaskConfig, UserTaskFormField, UserTaskFormFieldType} from '@process-engine/management_api_contracts';
```
Comment (Vyperus) [inline src/modules/form-widget/form-widget.ts]: Please sort this imports like mentioned above.

File: src/modules/confirm-widget/confirm-widget.ts (hunk 2 above)
Comment (Vyperus) [reply src/modules/confirm-widget/confirm-widget.ts]: done

File: src/modules/confirm-widget/confirm-widget.ts (hunk 1 above)
Comment (cmg-dev) [reply src/modules/confirm-widget/confirm-widget.ts]: @alexanderkasten please fix something like this right away... 

we don't want to whait for @SteffenKn to fix this.

File: src/modules/confirm-widget/confirm-widget.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  private _getAllOtherFormFields(formField: UserTaskFormField): Array<UserTaskFormField> {
+    const booleanFormFieldIndex: number = this.userTaskConfig.formFields.indexOf(formField);
+
+    const otherFormFields: Array<UserTaskFormField> = this.userTaskConfig.formFields.slice();
+    otherFormFields.splice(booleanFormFieldIndex, 1);
+
+    return otherFormFields;
+  }
```
Comment (cmg-dev) [inline src/modules/confirm-widget/confirm-widget.ts]: We can do this way better. This is the work of a `filter`

```ts
const  filteredFormFields: Array<UserTaskFormField> =  this.userTaskConfig.formFields.filter(field => field !== formField);
```

This should do the job! and make the whole method obsolete and is way more readable.

File: src/modules/confirm-widget/confirm-widget.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  private _getBooleanFormField(): UserTaskFormField {
```
Comment (cmg-dev) [inline src/modules/confirm-widget/confirm-widget.ts]: same here. a `filter` is the way we want it!

File: src/modules/confirm-widget/confirm-widget.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      const errorMessage: string = 'Confirm User Tasks must have a form field of type boolean that should get confirmed.';
+
+      this._notificationService.showNotification(NotificationType.ERROR, errorMessage);
+    }
+
+    this.confirmMessage = booleanFormField.label;
+    this.formFields = this._getAllOtherFormFields(booleanFormField);
+  }
+
+  public getFieldControl(field: UserTaskFormField): string {
```
Comment (cmg-dev) [inline src/modules/confirm-widget/confirm-widget.ts]: 1. Add a TODO.

```ts
//TODO: Move this to a UserTaskControlFactory
public getFieldControl(field: UserTaskFormField): string {
```

2. Create an issue for this.

**Comment by cmg-dev** [comment]
@Vyperus @alexanderkasten 

Please take care of my notes. And implement this carefully.

**Comment by cmg-dev** [comment]
@Vyperus Rebase onto develop and merge.

**Pull request merged**